### PR TITLE
Feat/rpc client ddt tasks

### PR DESCRIPTION
**Pull request by tangrammer**
Relates #376 to be merged in https://github.com/kubelt/kubelt/tree/feat/rpc-client
### `ddt rpc `
- [x] add `call` command for calling provider RPC endpoints
    
```
node ddt.js rpc call :kb:ping  -w default  -x value_1=1 -x value_2=2
Enter your password:  ******

com.kubelt.ddt.js default
:method kb_ping :params {:value_1 1, :value_2 2}
call result: pong
```

- [x] add `list` command for listing available RPC endpoints

```
> node ddt.js rpc list -w default

Enter your password:  ******
#{[:kb :auth] [:kb :ping] [:kb :pong] [:kb :auth :verify]}
```

### CODE REVIEW

File: src/main/com/kubelt/ddt/cmds/sdk/core/authenticate.cljs [hunk 1]
```diff
@@ -13,6 +13,32 @@
    [com.kubelt.sdk.v1 :as sdk]
    [com.kubelt.sdk.v1.core :as sdk.core]))
 
+(defn authenticate [args-map password on-authenticate]
```
Comment (tangrammer) [inline src/main/com/kubelt/ddt/cmds/sdk/core/authenticate.cljs:16]: refactor: just extracting logic outside so we can reuse/call it from other places

File: src/main/com/kubelt/lib/p2p.cljs [hunk 2]
```diff
@@ -14,35 +15,55 @@
...
-    (str (name scheme) "://" host ":" port path)))
+        path (cstr/join "" ["/@" core "/jsonrpc"])
+        uri (str (name scheme) "://" host ":" port path)]
+    (log/debug {::connection-uri uri})
+    uri))
+
+(defn- json-rpc-provider [sys core]
+  (JsonRpcProvider. (connection-uri sys core)))
+
+(defn- send
```
Comment (tangrammer) [inline src/main/com/kubelt/lib/p2p.cljs:26]: more refactors and so far all json-rpc logic is being concentrated here but feel free to move it to proper ns

File: src/main/com/kubelt/ddt/cmds/rpc/call.cljs [hunk 3]
```diff
@@ -20,10 +24,9 @@
                                 :describe "An RPC parameter (<name>=<value>)"
                                 :requiresArg true
                                 :demandOption "param name and value are required"
-                                :string true
+                                :array true
```
Comment (tangrammer) [inline src/main/com/kubelt/ddt/cmds/rpc/call.cljs:27]: do we agree on use this type so we can send more than 1 args?
each one should be prefixed ...
eg: `-x value_1=1 -x value_2=2`

Comment (crimeminister) [reply src/main/com/kubelt/ddt/cmds/rpc/call.cljs:28]: I did some experimentation and thought I remembered multiple parameter values being collected even when I had `:string true` set, but I just did a little test and it blew up. Looks like this change is necessary, thanks.

File: src/main/com/kubelt/ddt/cmds/sdk/core/authenticate.cljs [hunk 4]
```diff
@@ -13,6 +13,32 @@
...
+ (let [app-name (get args-map :app-name)
+       wallet-name (get args-map :wallet)]
+   (println app-name wallet-name)
```
Comment (crimeminister) [inline src/main/com/kubelt/ddt/cmds/sdk/core/authenticate.cljs:19]: Should we remove this `(println)`?

Comment (tangrammer) [reply src/main/com/kubelt/ddt/cmds/sdk/core/authenticate.cljs]: yep!

File: src/main/com/kubelt/ddt/cmds/sdk/core/authenticate.cljs (hunk 1 above)
Comment (crimeminister) [reply src/main/com/kubelt/ddt/cmds/sdk/core/authenticate.cljs:16]: Sounds good!

File: src/main/com/kubelt/lib/p2p.cljs (hunk 2 above)
Comment (crimeminister) [reply src/main/com/kubelt/lib/p2p.cljs:26]: After this lands I will look at moving the RPC client into the SDK state map, and we can re-use it that way. That'll give us the option of injecting an RPC client implementation as well, so we can experiment with different options there (at a minimum, choose between `JsonRpcProvider` and our local client).

**Review by crimeminister — APPROVED** [review]

**Pull request merged**